<launch>
  <rosparam file="$(find arcsys2_control)/config/arcsys2_control.yaml" command="load" ns="/arcsys2" />

  <node name="arcsys2_control" pkg="arcsys2_control" type="arcsys2_control_node" output="screen" ns="/arcsys2">
    <rosparam file="$(find arcsys2_control)/config/real_config.yaml" command="load" />
  </node>

  <node name="controller_spawner" pkg="controller_manager" type="spawner"
        respawn="false" output="screen" ns="/arcsys2"
        args="joint_state_controller
              rail_to_shaft_position_controller
              shaft_to_arm0_position_controller
              arm0_to_arm1_position_controller
              arm1_to_arm2_position_controller
              arm2_to_effector_base_position_controller
              effector_base_to_effector_end_position_controller" />

  <node name="robot_state_publisher" pkg="robot_state_publisher"
        type="robot_state_publisher" respawn="false" output="screen">
    <remap from="/joint_states" to="/arcsys2/joint_states" />
  </node>
</launch>
